### DISCUSSION TITLE
How to update data in the running Cube?

### DISCUSSION THREAD

**Discussion by tht-activeviam** [atoti/atoti#486] [Q&A] [answered]
We have an atoti app running with 4/5 cubes. One of the cube get the data from the database. Restarting the app will take about 45 minutes.
How to update the related cube without restarting the app when new data arrives in the database?

**Comment by bloussou** [comment]
Hello,
Are you looking for these methods?

- [`load_csv()`](https://docs.atoti.io/0.6.4/lib/atoti.table.html#atoti.table.Table.load_csv)
- [`load_pandas()`](https://docs.atoti.io/0.6.4/lib/atoti.table.html#atoti.table.Table.load_pandas)
- [`load_parquet()`](https://docs.atoti.io/0.6.4/lib/atoti.table.html#atoti.table.Table.load_parquet)
- [`append()`](https://docs.atoti.io/0.6.4/lib/atoti.table.html#atoti.table.Table.append)
- and more...

**Reply by tht-activeviam** [reply]
We are using `session.create_table(xxx)`, then call `table.load_pandas()`. I guess we can create another python function to invoke restarting that cube. What's the recommended way to expose that function(eg. REST api?) on the running app? Atoti is launched using `python -m app` command.

**Comment by bloussou** [comment, marked as answer]
1. When executing `load_*()` on a table, cubes based on this table are automatically updated (no need to care about this).
2. For an app in production you can create [`endpoints`](https://docs.atoti.io/latest/lib/atoti.session.html#atoti.session.Session.endpoint) to execute loading when needed. Some other ways: 
    - Setup a file watcher as described in [this **how to**](https://docs.atoti.io/latest/how_tos/watch/file_watching.html).
    - Setup polling as done in the [project template](https://github.com/atoti/project-template/blob/b811f21cf67162c0b0692fe223c37f4192669caa/app/app.py#L21-L24).

**Reply by tht-activeviam** [reply]
Thanks for the links. I concluded I have two approaches:

1. UI widget
Create a UI widget to reload the cube. User have to manually trigger it.

2. API endpoint
Create an API endpoint to reload the cube. [Create a service account](https://docs.atoti.io/latest/how_tos/security/security.html#Technical-users-and-service-accounts) to authenticate the endpoint. Create a batch job to call the endpoint on scheduled time slot. 

Thanks!

**Reply by bloussou** [reply]
You don't have to reload the cube **it is done automatically**. Why do you want to do it manually?

**Reply by tht-activeviam** [reply]
Yes, I just need to call `table.load_pandas()`.
We don't know when will the new data arrives in database. So either we can manually trigger it or have a scheduled job to call it nightly.
